network.tf

```diff
@@ -0,0 +1,91 @@
+locals {
+    public_cidr = ["10.0.0.0/24", "10.0.1.0/24"]
+    private_cidr = ["10.0.10.0/24", "10.0.11.0/24"]
+}
+
+# VPC
+resource "aws_vpc" "main" {
+  cidr_block = "10.0.0.0/16"
+}
+
+# subnet
+resource "aws_subnet" "public" {
+
+  count = 2
+
+  vpc_id     = aws_vpc.main.id
+  cidr_block = local.public_cidr[count.index]
+
+  tags = {
+    Name = "public${count.index}"
+  }
+}
+
+resource "aws_subnet" "private" {
+  count = 2
```

Comment:
Good job with using `count` here.

However, the task says `at least two subnets` this means that we might encounter a situation where 3 subnets per layer are required or even 5

You need to create more universal approach that allows setting `count` value dynamically:
https://www.terraform.io/docs/language/functions/length.html